scripts/gz10/build_parent_sample.py

```diff
@@ -0,0 +1,60 @@
+import h5py
+import healpy as hp
+import numpy as np
+import os
+from tqdm import tqdm
+import argparse
+from multiprocessing import Pool
+
+def process_index(args):
```

Comment:
Generally this code needs docstrings, but to prioritise, this function is probably the most important to explain